iamdata.tf

```diff
@@ -0,0 +1,178 @@
+data "aws_iam_policy_document" "cross_account_assume_role_policy_mfa" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = [
+        "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
+      ]
+    }
+
+    actions = ["sts:AssumeRole"]
+
+    condition {
+      test     = "Bool"
+      variable = "aws:MultiFactorAuthPresent"
+      values = [
+        "true",
+      ]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "iam_restricted_admin" {
+  statement {
+    sid = "AllowFullAdminExceptSomeWithMFA"
+    not_actions = [
+      "logs:Delete*",
+      "cloudtrail:Delete*",
+      "cloudtrail:Stop*",
+      "cloudtrail:Update*",
+      "sts:AssumeRole",
+      "sts:AssumeRoleWithSAML",
+      "sts:AssumeRoleWithWebIdentity",
+      "sts:GetFederationToken",
+    ]
+    resources = [
+      "*",
+    ]
+  }
+  statement {
+    sid = "AllowFullAdminExceptSomeS3WithMFA"
+    not_actions = [
+      "s3:Delete*",
+      "s3:Put*",
+    ]
+    resources = var.s3_bucket_paths_to_protect
+  }
+}
+
+data "aws_iam_policy_document" "role_assumption" {
+  statement {
+    sid = "AssumeRole"
+
+    actions = [
+        "sts:AssumeRole"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "${aws_iam_role.this.arn}"
+    ]
+
+    condition {
+      test     = "BoolIfExists"
```

Comment:
I think you want `Bool` instead of `BoolIfExists`
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html#condition-keys-multifactorauthpresent

Reply:
@rchildress87 That document says BoolIfExists is correct, the policy we're using is the AWS published one that everyone uses.

We recommend that you use the [BoolIfExists](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_IfExists) operator to check whether a request is authenticated using MFA.

Reply:
I think the `bool` example under Alternative Combinations should be considered as better option but, according to above documentation, this does emulate the policy that is created when you use the IAM visual editor and choose *MFA Required*. It also matches the policy that this data source is replacing. I'll approve.